src/App.js

```diff
@@ -1,41 +1,50 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Route } from 'react-router-dom';
 import data from './data';
 
 // Components
 import Navigation from './components/Navigation';
 import Products from './components/Products';
 import ShoppingCart from './components/ShoppingCart';
+import { ProductContext } from "./contexts/ProductContext";
+import { CartContext } from "./contexts/CartContext";
 
 function App() {
 	const [products] = useState(data);
 	const [cart, setCart] = useState([]);
 
+//	const [cart, setCart] = useState(localStorage.getItem("my cart items") || []);
+
 	const addItem = item => {
 		setCart([...cart, item]);
 	};
 
+	//local storage shenanigans
+//	useEffect(() => {
+//	  localStorage.setItem("my cart items", cart);
+//	})
+//	let retrieveState = 
+
 	return (
-		<div className="App">
-			<Navigation cart={cart} />
-
-			{/* Routes */}
-			<Route
-				exact
-				path="/"
-				render={() => (
-					<Products
-						products={products}
-						addItem={addItem}
+		<ProductContext.Provider value={{products, addItem}}>
+			<CartContext.Provider value={{cart}}>
+				<div className="App">
+					<Navigation cart={cart} />
+
+					{/* Routes */}
+					<Route
+						exact
+						path="/"
+						component={Products}
+					/>
+
+					<Route
+						path="/cart"
+						render={() => <ShoppingCart cart={cart} />}
 					/>
```

Comment:
We do not need to pass the cart component to this `Route` since we will be able to pull `cart` from the `CartContext.Provider`.  We can just pass the `ShoppingCart` component to the `Route` component the same way we did for `Products`.